src/index/base.h

```diff
@@ -78,6 +78,17 @@ class BaseIndex : public CValidationInterface
 
     virtual bool AllowPrune() const = 0;
 
+    virtual DB& GetDB() const = 0;
+
+    /// Write update index entries for a newly connected block.
+    virtual bool WriteBlock(const CBlock& block, const CBlockIndex* pindex) { return true; }
+
+    /// Update the internal best block index as well as the prune lock.
+    void SetBestBlockIndex(const CBlockIndex* block);
+
+    /// Get the name of the index for display in logs.
```

Comment:
I'm kind of confused by this; it's (like `GetDB()`) a method that child classes are supposed to override. What does it mean for it to be private?

Reply:
Thanks for reviewing! It looks to me that a base class virtual method doesn't need to be accessible to be overridden, but it can only be `private` if fully overridden, e.g. no inherited base implementation is used by the child classes.

References:

- https://en.cppreference.com/w/cpp/language/virtual

- "An implementation member should be protected if it provides an operation or data that a derived class will need to use in its own implementation. Otherwise, implementation members should be private." - C++ Primer (Lippman, Lajoie, Moo)

Reply:
Updated the OP with this context.

Reply:
Thanks for looking it up! I have no objection to doing it this way, but it's a dusty corner of C++ to me.

We might want to move the methods that are intended to (or, must) be overridden by child classes together in the class declaration, for documentation purposes.

Reply:
> Thanks for looking it up! I have no objection to doing it this way, but it's a dusty corner of C++ to me.
> 
> We might want to move the methods that are intended to (or, must) be overridden by child classes together in the class declaration, for documentation purposes.

Done!